### DISCUSSION TITLE
Leash/Attach Entity to another Entity.

### DISCUSSION THREAD

**Discussion by mikuhl-dev** [Minestom/Minestom#739] [Help]
You can use AttachEntityPacket, but it seems like the timing on when you can send this is peculiar, and how to ensure new players seeing the entity also get sent this packet. I have tried in `updateNewViewer` and `spawn` with no success. Seems to work randomly however, which alludes to some race condition.

**Comment by emortaldev** [comment]
If you're leashing the entity the moment they're spawned, try setInstance.thenRun(() -> ...)

**Reply by mikuhl-dev** [reply]
I think your solution would work for people who are already viewing the entities, but not a new viewer, sending the packet after one tick seemed to do the trick like this:
```java
    @Override
    public void updateNewViewer(@NotNull Player player) {
        super.updateNewViewer(player);
        player.scheduleNextTick(e -> player.sendPacket(new AttachEntityPacket(this, holder)));
    }
```
But this does not account for if only ONE of the entities are being viewed, which is probably quite unlikely as leashed entities are usually next to eachother.
